projects/storefrontlib/src/cms-components/product/product-tabs/product-reviews/product-reviews.component.ts

```diff
@@ -85,7 +79,21 @@ export class ProductReviewsComponent {
     this.reviewForm.controls.rating.setValue(rating);
   }
 
-  submitReview(product: Product): void {
+  private markFormAsTouched(): void {
+    Object.keys(this.reviewForm.controls).forEach(key => {
+      this.reviewForm.controls[key].markAsTouched();
+    });
+  }
+
+  submitReview(product: Product) {
+    if (this.reviewForm.valid) {
+      this.addReview(product);
+    } else {
+      this.markFormAsTouched();
```

Comment:
we can just do `this.reviewForm.markAsUntouched();`

Reply:
Same as above.

Reply:
The above works as follows: when form is empty (invalid), we don't submit and mark it as touched, so the red border can appear on them; if it's valid, we submit (and send the actual request to the backend). This is a follow up to the removal of [disabled] attributes from submit form buttons.